packages/api/src/modules/providers/demarchesSimplifiees/demarchesSimplifiees.service.ts

```diff
@@ -68,13 +71,15 @@ export class DemarchesSimplifieesService
         );
     }
 
-    async getDemandeSubventionBySiren(siren: Siren): Promise<DemandeSubvention[] | null> {
-        const demandes = await demarchesSimplifieesDataRepository.findBySiren(siren);
-        return this.entitiesToSubventions(demandes);
-    }
+    async getDemandeSubvention(id: StructureIdentifier): Promise<DemandeSubvention[]> {
+        const demandes: DemarchesSimplifieesDataEntity[] = [];
+
+        if (id instanceof EstablishmentIdentifier && id.siret) {
+            demandes.push(...(await demarchesSimplifieesDataRepository.findBySiret(id.siret)));
+        } else if (id instanceof AssociationIdentifier && id.siren) {
+            demandes.push(...(await demarchesSimplifieesDataRepository.findBySiren(id.siren)));
+        }
```

Comment:
```suggestion
        let demandes: DemarchesSimplifieesDataEntity[] = [];

        if (id instanceof EstablishmentIdentifier && id.siret) {
            demandes = await demarchesSimplifieesDataRepository.findBySiret(id.siret);
        } else if (id instanceof AssociationIdentifier && id.siren) {
            demandes = await  demarchesSimplifieesDataRepository.findBySiren(id.siren);
        }
```
pourquoi les push ? ça a l'air bien compliqué de destructurer les tableaux

Reply:
Tout à fait

Reply:
Done